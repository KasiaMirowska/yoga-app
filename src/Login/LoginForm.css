  .login{
    width: 30%;
    background-color: rgb(233, 224, 231); 
    text-align: center; 
    padding-top: 30px; 
    padding-bottom: 50px;
    margin: auto;
    margin-bottom: 40px;
    filter: drop-shadow(0 0 0.75rem rgb(151, 132, 136));
  }
  
  input {
    width: 70%; 
    font-family: inherit;
    border: 0;
    border-bottom: 2px solid gray;
    outline: 0;
    font-size: 1rem;
    color: grey;
    padding: 7px 0;
    /* display: inline; */
    background: transparent;  
    transition: border-color 0.2s;
    /* margin-right: 15px;  */
  }
  placeholder {
    color: transparent; 
  }
  .login-button {
    margin: 20px;
    border-radius: 5px;
    
  } 
  .section {
      margin: 20px;
  }