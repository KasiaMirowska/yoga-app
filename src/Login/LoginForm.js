import React from 'react';
import AuthCalls from '../services/Auth_service';
import TokenService from '../services/token-service';
import YogaContext from '../Context';
import './LoginForm.css';

export default class LoginForm extends React.Component {
    static contextType = YogaContext;

    state = {
        error: null,
    }

    handleLogin = (e) => {
        e.preventDefault();
        const { userName, password } = e.target

        const credentials = {
            userName: userName.value,
            password: password.value,
        }


        AuthCalls.postLogin(credentials)
            .then(res => {
                userName.value = ''
                password.value = ''
                TokenService.saveAuthToken(res.authToken)
                this.props.onLoginSuccess()
            })
            .catch(res => {
                console.log(res)
                this.setState({
                    error: res
                })
            })
    }

    handleAfterErrorDisplay = ()=> {
        this.setState({
            error: null
        })
    }
    handleUsenNameCleanUp = (e) => {
        e.preventDefault();
        let userName = e.target;
        if(userName.value !== null) {
            userName.value = '';
            this.setState({
                error: null
            })
        } 
    }

    handlePasswordCleanUp = (e) => {
        e.preventDefault();
        let password = e.target;
        if(password.value !== null) {
           password.value = '';
           this.setState({
            error: null
        })
        }  
    }
    render() {
        return (
            <div className='login'>
                <div className='error'>
                { this.state.error ? this.state.error.message : null }
                </div>
                <form onSubmit={this.handleLogin}>
                    <div className='section'>
                        <input className='inp' type='text' id='userName' placeholder='username' onClick={this.handleUsenNameCleanUp}>
                        </input>
                    </div>
                    <div className='section'>
                        <input className='inp' type='password' placeholder='password' id='password' onClick={this.handlePasswordCleanUp}>
                        </input>
                    </div>
                    <button className='login-button' type="reset" onClick={this.handleAfterErrorDisplay}>Reset</button>
                    <button className='login-button' type='submit'>Login</button>
                </form>
            </div>
        )
    }
}