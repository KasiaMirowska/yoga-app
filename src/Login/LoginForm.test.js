import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';
import {BrowserRouter as BR} from 'react-router-dom';

describe('LoginForm component', () => {
    const props = {match: {params: {path: '/login'}}}
    
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><LoginForm {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const loginPage = renderer.create(<BR><LoginForm/></BR>);
        expect(loginPage.toJSON()).toMatchSnapshot();
    })

})