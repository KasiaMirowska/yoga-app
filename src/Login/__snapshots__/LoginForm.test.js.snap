// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="userName"
    >
      Username
    </label>
    <input
      id="userName"
      type="text"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      type="text"
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
